jsonargparse_tests/typehints_tests.py

```diff
@@ -226,6 +227,44 @@ def test_complex_number(self):
         self.assertEqual(parser.dump(cfg), 'complex: (2+3j)\n')
 
 
+    def test_list_append(self):
+        parser = ArgumentParser(error_handler=None)
+        parser.add_argument('--val', type=Union[int, str, List[int]])
+        self.assertEqual(0, parser.parse_args(['--val+=0']).val)
+        self.assertEqual([1, 2, 3], parser.parse_args(['--val+=1', '--val+=2', '--val+=3']).val)
+        self.assertEqual([1, 2, 3], parser.parse_args(['--val=[1,2]', '--val+=3']).val)
+        self.assertEqual(1, parser.parse_args(['--val=a', '--val+=1']).val)
+        with warnings.catch_warnings(record=True) as w:
+            self.assertEqual(3, parser.parse_args(['--val=[1,2]', '--val=3']).val)
+            self.assertIn('Replacing list value "[1, 2]" with "3"', str(w[0].message))
+
+
+    def test_list_append_config(self):
+        parser = ArgumentParser(error_handler=None)
+        parser.add_argument('--cfg', action=ActionConfigFile)
+        parser.add_argument('--val', type=List[int], default=[1, 2])
+        self.assertEqual([3, 4], parser.parse_args(['--cfg', 'val: [3, 4]']).val)
+        self.assertEqual([1, 2, 3], parser.parse_args(['--cfg', 'val+: 3']).val)
+        self.assertEqual([1, 2, 3, 4], parser.parse_args(['--cfg', 'val+: [3, 4]']).val)
```

Comment:
You could also add tests that interleave setting init_args as in the tests removed in https://github.com/PyTorchLightning/pytorch-lightning/pull/13129